--- 
title: "Pathways" 
subtitle: "CBW workshop"
author: "Gary Bader, Veronique Voisin, Ruth Isserlin, Robin Haw, Michael Hoffman"
date: "last modified `r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
github-repo: rstudio/bookdown-demo
favicon: images/favicon.ico
description: "Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information. "
---

# Pathway and Network Analysis of -Omics Data (May 2021)

The CBW has developed a 2.5 day course covering the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information. The workshop focuses on the principles and concepts required for analyzing and conducting pathway and network analysis on a gene list from any organism, although focus will be on human and model eukaryotic organisms.

## Meet your Faculty

### Gary Bader
Principal Investigator, University of Toronto 

<img src="./images/faculty/gary_bader.jpeg" style="float:left;width:125px;padding-right:10px;" />
Dr. Bader develops biological network analysis and pathway information resources. He created the Biomolecular Interation Network Database ( [BIND](http://bind.ca) ) while working on his PhD and currently helps lead development of the free Cytoscape network visualization and analysis software [Cytoscape](http://cytoscape.org/). 

### Robin Haw
Manager of Reactome Outreach and Scientific Associate, OICR

<img src="./images/faculty/robin_haw.jpeg" style="float:right;width:125px;padding-left:10px;" />
Robin is a microbiologist with experience in bioinformatics, curation and outreach. Dr. Haw has a PhD in genetics and was a senior curator at Biomolecular Interaction Network Database (BIND) and managing curator at Science Signaling’s Signal Transduction Knowledge Environment (STKE). He has been responsible for coordinating outreach, Reactome presentations and training at workshops and conferences.

### Michael Hoffman
Scientist, Princess Margaret Cancer Centre Assistant Professor, Departments of Medical Biophysics and Computer Science, University of Toronto Faculty Affiliate, Vector Institute 

<img src="./images/faculty/michael_hoffman.jpeg" style="float:right;width:125px;padding-left:10px;" />
Michael Hoffman creates predictive computational models to understand interactions between genome, epigenome, and phenotype in human cancers. He implemented the genome annotation method Segway, which simplifies interpretation of large multivariate genomic datasets, and was a linchpin of the NIH ENCODE Project analysis. He was named a CIHR New Investigator and has received several awards for his academic work, including the NIH K99/R00 Pathway to Independence Award, and the Ontario Early Researcher Award. 

### Veronique Voisin
Research Associate, Donnelly Centre for Cellular and Biomolecular Research, University of Toronto 

<img src="./images/faculty/veronique_voisin.jpeg" style="float:left;width:125px;padding-right:10px;" />
Veronique is currently a bioinformatician applying pathway and networks analysis to high throughput genomics data for OICR cancer stem cell program. Previously, she worked on characterizing the gene signatures of different types of leukemias using a murine model

&nbsp;
&nbsp;

### Ruth Isserlin
Research data analyst, Donnelly Centre for Cellular and Bimolecular Research, University of Toronto

<img src="./images/faculty/ruth_isserlin.jpeg" style="float:right;width:125px;padding-left:10px;" />
Bioinformatician and data analyst in the Bader lab applying pathway and data analysis to varied data types. Developed Enrichment Map App for Cytoscape, an app to visually translate functional enrichment results from popular enrichment tools like GSEA to networks. Further developed the Enrichment Map Pipeline including development of additional Apps to help summarize and analyze resulting Enrichment Maps, including PostAnalysis, WordCloud, and AutoAnnotate App. 


***

Thank you for attending the Pathway and Network Analysis of Omics Data workshop! Help us make this workshop better by filling out [our survey](https://forms.gle/D8w8qyJ1r71rFnZe9).

*** 

```{r include=FALSE}
# automatically create a bib database for R packages
knitr::write_bib(c(
  .packages(), 'bookdown', 'knitr', 'rmarkdown'
), 'packages.bib')
```

**This work is licensed under a [Creative Commons Attribution-ShareAlike 3.0 Unported License](http://creativecommons.org/licenses/by-sa/3.0/deed.en_US). This means that you are able to copy, share and modify the work, as long as the result is distributed under the same license.**

Icons are from the [“Very Basic. Android L Lollipop” set by Ivan Boyko](https://www.iconfinder.com/iconsets/very-basic-android-l-lollipop) licensed under [CC BY 3.0](https://creativecommons.org/licenses/by/3.0/) and [Icons8](icons8.com). 
