<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Module 3 Lab: g:profiler Visualization | Pathways</title>
  <meta name="description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Module 3 Lab: g:profiler Visualization | Pathways" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Module 3 Lab: g:profiler Visualization | Pathways" />
  
  <meta name="twitter:description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  

<meta name="author" content="Gary Bader, Veronique Voisin, Ruth Isserlin, Robin Haw, Michael Hoffman" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="module-3-network-visualization-and-analysis-with-cytoscape.html"/>
<link rel="next" href="module-4-in-depth-analysis-of-networks-and-pathways.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>




<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CBW Pathways Workshop</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Pathway and Network Analysis of -Omics Data ( May 2021 )</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#meet-your-faculty"><i class="fa fa-check"></i>Meet your Faculty</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#gary-bader"><i class="fa fa-check"></i>Gary Bader</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#robin-haw"><i class="fa fa-check"></i>Robin Haw</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#michael-hoffman"><i class="fa fa-check"></i>Michael Hoffman</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#veronique-voisin"><i class="fa fa-check"></i>Veronique Voisin</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#ruth-isserlin"><i class="fa fa-check"></i>Ruth Isserlin</a></li>
</ul></li>
</ul></li>
<li><a href="welcome.html#welcome">Welcome <a id="welcome"></a></a><ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html"><i class="fa fa-check"></i>Class Materials</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#schedule"><i class="fa fa-check"></i>Workshop Schedule</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop"><i class="fa fa-check"></i>Pre-Workshop Materials and Laptop Setup Instructions</a><ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#laptop-setup-instructions"><i class="fa fa-check"></i>Laptop Setup Instructions</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#basic-programs"><i class="fa fa-check"></i>Basic programs</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#java-11-installation"><i class="fa fa-check"></i>Java 11 Installation</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#cytoscape-installation"><i class="fa fa-check"></i>Cytoscape Installation</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#gsea-installation"><i class="fa fa-check"></i>GSEA Installation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop-tutorials"><i class="fa fa-check"></i>Pre-workshop Tutorials</a><ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#cytoscape-preparation-tutorials"><i class="fa fa-check"></i>Cytoscape Preparation tutorials</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop-readings-and-lectures"><i class="fa fa-check"></i>Pre-workshop Readings and Lectures</a></li>
<li><a href="welcome.html#additional-tutorials">Additional tutorials <a id="additionaltutorials"></a></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i>Module 1 - Introduction to Pathway and Network Analysis (Gary Bader)</a></li>
<li class="chapter" data-level="" data-path="module-2-finding-over-represented-pathways-veronique-voisin.html"><a href="module-2-finding-over-represented-pathways-veronique-voisin.html"><i class="fa fa-check"></i>Module 2: Finding Over-represented Pathways (Veronique Voisin)</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html"><i class="fa fa-check"></i>Module 2 lab - g:Profiler</a><ul>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#introduction"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#goal-of-the-exercise-1"><i class="fa fa-check"></i>Goal of the exercise 1</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#data"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#exercise-1"><i class="fa fa-check"></i>Exercise 1 - run g:Profiler</a><ul>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-1---launch-gprofiler."><i class="fa fa-check"></i>Step 1 - Launch g:Profiler.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-2---input-query"><i class="fa fa-check"></i>Step 2 - input query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-3---adjust-parameters."><i class="fa fa-check"></i>Step 3 - Adjust parameters.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-4---run-query"><i class="fa fa-check"></i>Step 4 - Run query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-5---explore-the-results."><i class="fa fa-check"></i>Step 5 - Explore the results.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-6-expand-the-stats-tab"><i class="fa fa-check"></i>Step 6: Expand the stats tab</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-7-save-the-results"><i class="fa fa-check"></i>Step 7: Save the results</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-8"><i class="fa fa-check"></i>Step 8</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#exercise-2-load-and-use-a-custom-.gmt-file-and-run-the-query"><i class="fa fa-check"></i>Exercise 2: Load and use a custom .gmt file and run the query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#optional-steps"><i class="fa fa-check"></i>Optional steps</a><ul>
<li><a href="gprofiler-lab.html#optional-1"><strong>Optional 1</strong>:</a></li>
<li><a href="gprofiler-lab.html#option-2"><strong>Option 2</strong>:</a></li>
<li><a href="gprofiler-lab.html#option-3"><strong>Option 3</strong> :</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html"><i class="fa fa-check"></i>Module 2 lab - GSEA</a><ul>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#introduction-1"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#goal-of-the-exercise"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#data-1"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#background"><i class="fa fa-check"></i>Background</a><ul>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#how-to-generate-a-rank-file."><i class="fa fa-check"></i>How to generate a rank file.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#start-the-exercise"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step1."><i class="fa fa-check"></i>Step1.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-2."><i class="fa fa-check"></i>Step 2.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step3."><i class="fa fa-check"></i>Step3.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-4."><i class="fa fa-check"></i>Step 4.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-5."><i class="fa fa-check"></i>Step 5.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#additional_information"><i class="fa fa-check"></i>Additional information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3-network-visualization-and-analysis-with-cytoscape.html"><a href="module-3-network-visualization-and-analysis-with-cytoscape.html"><i class="fa fa-check"></i>Module 3: Network Visualization and Analysis with Cytoscape</a><ul>
<li class="chapter" data-level="" data-path="module-3-network-visualization-and-analysis-with-cytoscape.html"><a href="module-3-network-visualization-and-analysis-with-cytoscape.html#module-3-lab"><i class="fa fa-check"></i>Module 3 Lab</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html"><i class="fa fa-check"></i>Module 3 Lab: g:profiler Visualization</a><ul>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#goal-of-the-exercise-2"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#data-2"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#enrichmentmap"><i class="fa fa-check"></i>EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#description-of-this-exercise"><i class="fa fa-check"></i>Description of this exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#start-the-exercise-1"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#exercise-1a"><i class="fa fa-check"></i>Exercise 1a</a><ul>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#step-1"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#step-2"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#step3-explore-the-results"><i class="fa fa-check"></i>Step3: Explore the results:</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#explore-detailed-results"><i class="fa fa-check"></i>Explore Detailed results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#exercise-1b"><i class="fa fa-check"></i>Exercise 1b</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#exercise-1c"><i class="fa fa-check"></i>Exercise 1c</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#goal-of-the-exercise-3"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#data-3"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#gsea-output-and-enrichmentmap"><i class="fa fa-check"></i>GSEA output and EnrichmentMap</a><ul>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#enrichmentmap-1"><i class="fa fa-check"></i>EnrichmentMap</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#start-the-exercise-2"><i class="fa fa-check"></i>Start the exercise</a><ul>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#step-1-1"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#step-2-1"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#step-3"><i class="fa fa-check"></i>Step 3</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#step-4"><i class="fa fa-check"></i>Step 4</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#step-5"><i class="fa fa-check"></i>Step 5</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#step-6"><i class="fa fa-check"></i>Step 6</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#step-7"><i class="fa fa-check"></i>Step 7</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#goal-of-the-exercise-4"><i class="fa fa-check"></i>Goal of the exercise:</a><ul>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#set-up"><i class="fa fa-check"></i>Set Up</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#step-1---launch-rstudio"><i class="fa fa-check"></i>Step 1 - launch RStudio</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#step-2---create-a-new-project"><i class="fa fa-check"></i>Step 2 - create a new project</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#step-3---open-example-up-rnotebook"><i class="fa fa-check"></i>Step 3 - Open example up RNotebook</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#step-4---define-notebook-parameters"><i class="fa fa-check"></i>Step 4 - Define Notebook parameters</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#step-5---step-through-notebook-to-run-the-analysis"><i class="fa fa-check"></i>Step 5 - Step through notebook to run the analysis</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#exercises"><i class="fa fa-check"></i>Exercises</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#additional-resources"><i class="fa fa-check"></i>Additional resources</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-4-in-depth-analysis-of-networks-and-pathways.html"><a href="module-4-in-depth-analysis-of-networks-and-pathways.html"><i class="fa fa-check"></i>Module 4: In-depth Analysis of Networks and Pathways</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html"><i class="fa fa-check"></i>Module 5: Regulatory Network Analysis</a><ul>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#lecture"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#practical-lab-1-chip_seq-data---great-and-meme-chip"><i class="fa fa-check"></i>Practical lab 1: chIP_seq data - GREAT and MEME-chIP</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#practical-lab-2-gene-list---iregulon"><i class="fa fa-check"></i>Practical lab 2: gene list - iREgulon</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#additional-slides-about-the-tools-segway-and-behst-presented-during-the-lecture"><i class="fa fa-check"></i>Additional slides about the tools Segway and BEHST presented during the lecture</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html"><i class="fa fa-check"></i>Module 5 Lab 1: Gene Regulation and Motif Analysis Practical Lab /chIP-seq</a><ul>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#goal-of-this-practical-lab"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#dataset-used-during-this-practical-lab"><i class="fa fa-check"></i>Dataset used during this practical lab</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#exercise-1---run-pathway-analysis-using-great"><i class="fa fa-check"></i>Exercise 1 - Run pathway analysis using GREAT</a><ul>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#perform-pathway-enrichment"><i class="fa fa-check"></i>Perform pathway enrichment</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#explore-the-results."><i class="fa fa-check"></i>Explore the results.</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#perform-pathway-enrichment---proximal-approach"><i class="fa fa-check"></i>Perform pathway enrichment - Proximal approach</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#explore-the-results.---proximal-analysis"><i class="fa fa-check"></i>Explore the results. - proximal analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#exercise-2---build-an-enrichment-map-to-visualize-great-results"><i class="fa fa-check"></i>Exercise 2 - Build an enrichment map to visualize GREAT results</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#exercise-3-optional-auto-annotate-the-distal-enrichment-map-build-the-proximal-map-and-compare-both-the-proximal-and-distal-maps"><i class="fa fa-check"></i>Exercise 3 (optional): Auto-annotate the distal enrichment map, build the proximal map and compare both the proximal and distal maps</a><ul>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#optional-exercise-3a-autoannotate-the-enrichment-map"><i class="fa fa-check"></i>Optional exercise 3a: AutoAnnotate the enrichment map:</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#optional-exercise-3b-repeat-the-process-of-building-an-enrichment-map-using-the-proximal-data-proximal_gobp_greatexportall.tsv."><i class="fa fa-check"></i>Optional exercise 3b: Repeat the process of building an enrichment map using the proximal data (Proximal_GOBP_greatExportAll.tsv).</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#optional-exercise-3c-repeat-the-process-by-building-both-the-proximal-and-distal-enrichment-maps-at-the-same-time."><i class="fa fa-check"></i>Optional exercise 3c: Repeat the process by building both the Proximal and Distal enrichment maps at the same time.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#exercise-4-add-runx1-targets-and-runx1-ko-up-and-down-regulated-genes-on-the-distal-enrichment-map-as-a-new-gene-set."><i class="fa fa-check"></i>Exercise 4: Add RUNX1 targets and RUNX1 KO up and down regulated genes on the distal enrichment map as a new gene-set.</a><ul>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#step-4a-add-the-.gmt-file-to-the-enrichment-map"><i class="fa fa-check"></i>step 4a: Add the .gmt file to the enrichment map:</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#step-4b-optional-change-the-edge-style-of-the-signature-gene-sets"><i class="fa fa-check"></i>Step 4b Optional: Change the edge style of the signature gene-sets:</a></li>
</ul></li>
<li><a href="regulatory-network-chipseq-lab.html#exercise-5-learning-how-to-run-meme-chip-from-the-meme-suite-httpsmeme-suite.orgmemetoolsmeme-chip">Exercise 5: Learning how to run MEME-chip from the MEME suite (<span>https://meme-suite.org/meme/tools/meme-chip</span>)</a><ul>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#exercise-5a-downloading-sequences-from-.bed-coordinates"><i class="fa fa-check"></i>Exercise 5a: Downloading sequences from .bed coordinates</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#exercise-5b-run-meme-chip"><i class="fa fa-check"></i>Exercise 5b: Run MEME-chIP</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#exercise-6-optional-get-the-iregulon-runx1-targets-and-find-the-mouse-orthologs-using-gorth-from-gprofiler-to-create-the-gmt-file-used-in-exercise-4."><i class="fa fa-check"></i>Exercise 6 (optional): Get the iRegulon RUNX1 targets and find the mouse orthologs using g:Orth (from g:Profiler) to create the gmt file used in Exercise 4.</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#end-of-lab"><i class="fa fa-check"></i>End of Lab</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html"><i class="fa fa-check"></i>Module 5 Lab 2: Gene Regulation and Motif Analysis Practical Lab / iRegulon</a><ul>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#iregulon-lab"><i class="fa fa-check"></i>iRegulon lab</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#goal"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#exercise-1.-detect-regulons-from-co-expressed-genes"><i class="fa fa-check"></i>Exercise 1. Detect regulons from co-expressed genes</a><ul>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#skills-learned-in-this-exercise"><i class="fa fa-check"></i>Skills learned in this exercise:</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#steps"><i class="fa fa-check"></i>Steps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#exercise-2.-create-a-metatargetome-using-iregulon-and-merge-2-networks-in-cytoscape."><i class="fa fa-check"></i>Exercise 2. Create a metatargetome using iRegulon and merge 2 networks in Cytoscape.</a><ul>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#steps-1"><i class="fa fa-check"></i>Steps</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#use-or-precomputed-iregulon-results"><i class="fa fa-check"></i>Use or precomputed iRegulon results:</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#notes-about-iregulon"><i class="fa fa-check"></i>Notes about iRegulon:</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#notes-about-cytoscape"><i class="fa fa-check"></i>Notes about Cytoscape:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#exercise-3.-use-enrichr-with-the-prostate-gene-list."><i class="fa fa-check"></i>Exercise 3. Use Enrichr with the prostate gene list.</a><ul>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#goal-1"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#steps-2"><i class="fa fa-check"></i>Steps</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#end-of-practical-lab"><i class="fa fa-check"></i>end of practical lab</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-6-gene-function-prediction.html"><a href="module-6-gene-function-prediction.html"><i class="fa fa-check"></i>Module 6: Gene Function Prediction</a></li>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html"><i class="fa fa-check"></i>Module 6 Lab: GeneMANIA (Cytoscape version)</a><ul>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html#goal-of-this-practical-lab-1"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html#exercise-1-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 1: QUESTIONS AND STEPS TO FOLLOW</a><ul>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html#steps-3"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html#exercise-1-answers"><i class="fa fa-check"></i>EXERCISE 1 ANSWERS</a></li>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html#exercise-2-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 2: QUESTIONS AND STEPS TO FOLLOW</a><ul>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html#steps-4"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html#exercise-3-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 3: QUESTIONS AND STEPS TO FOLLOW</a><ul>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html#steps-5"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html#some-definitions"><i class="fa fa-check"></i>SOME DEFINITIONS:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html"><i class="fa fa-check"></i>Module 6 Lab: GeneMANIA (web version)</a><ul>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#goal-of-this-practical-lab-2"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1-questions-and-steps-to-follow-1"><i class="fa fa-check"></i>EXERCISE 1: QUESTIONS AND STEPS TO FOLLOW</a><ul>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#steps-6"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1-answers-detailed-explanation-and-screenshots"><i class="fa fa-check"></i>EXERCISE 1 ANSWERS: DETAILED EXPLANATION AND SCREENSHOTS</a><ul>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1---steps-1-4"><i class="fa fa-check"></i>EXERCISE 1 - STEPS 1-4</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1---step-5"><i class="fa fa-check"></i>EXERCISE 1 - STEP 5</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1---step-6"><i class="fa fa-check"></i>EXERCISE 1 - STEP 6</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1---step-7"><i class="fa fa-check"></i>Exercise 1 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1---step-8"><i class="fa fa-check"></i>Exercise 1 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1---step-9"><i class="fa fa-check"></i>Exercise 1 - STEP 9</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1---step-10-layouts"><i class="fa fa-check"></i>Exercise 1 - STEP 10 (layouts)</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1---step-11-save-an-image"><i class="fa fa-check"></i>Exercise 1 - STEP 11 (save an image)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2-questions-and-steps-to-follow-1"><i class="fa fa-check"></i>EXERCISE 2: QUESTIONS AND STEPS TO FOLLOW</a><ul>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#steps-7"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2-answers-detailed-steps-and-screenshots"><i class="fa fa-check"></i>EXERCISE 2 ANSWERS: DETAILED STEPS AND SCREENSHOTS</a><ul>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---steps-1-to-4"><i class="fa fa-check"></i>Exercise 2 - STEPS 1 to 4</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---step-5"><i class="fa fa-check"></i>Exercise 2 - STEP 5</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---step-6."><i class="fa fa-check"></i>Exercise 2 - STEP 6.</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---step-7"><i class="fa fa-check"></i>Exercise 2 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---step-8"><i class="fa fa-check"></i>Exercise 2 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---step-9"><i class="fa fa-check"></i>Exercise 2 - STEP 9</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---step-10"><i class="fa fa-check"></i>Exercise 2 - STEP 10</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---step-11"><i class="fa fa-check"></i>Exercise 2 - STEP 11</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---step-12"><i class="fa fa-check"></i>Exercise 2 - STEP 12</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---step-13."><i class="fa fa-check"></i>Exercise 2 - STEP 13.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-3-questions-and-steps-to-follow-1"><i class="fa fa-check"></i>EXERCISE 3: QUESTIONS AND STEPS TO FOLLOW</a><ul>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#steps-8"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-3-more-details-and-screenshots"><i class="fa fa-check"></i>Exercise 3: MORE DETAILS AND SCREENSHOTS</a><ul>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-3---steps-1---3"><i class="fa fa-check"></i>Exercise 3 - STEPS 1 - 3</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-3---step-4"><i class="fa fa-check"></i>Exercise 3 - STEP 4</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-3---step-5"><i class="fa fa-check"></i>Exercise 3 - STEP 5</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-3---steps-6"><i class="fa fa-check"></i>Exercise 3 - STEPS 6</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-3---step-7"><i class="fa fa-check"></i>Exercise 3 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-3---step-8"><i class="fa fa-check"></i>Exercise 3 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-3---step-9"><i class="fa fa-check"></i>Exercise 3 - STEP 9</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#some-definitions-1"><i class="fa fa-check"></i>SOME DEFINITIONS:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html"><i class="fa fa-check"></i>Integrated Assignment</a><ul>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#final-slides"><i class="fa fa-check"></i>Final slides</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#goal-2"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#dataset-1"><i class="fa fa-check"></i>DATASET 1</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#background-1"><i class="fa fa-check"></i>Background</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#data-processing"><i class="fa fa-check"></i>Data processing</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#part-1-run-gprofiler"><i class="fa fa-check"></i>PART 1: run g:Profiler</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#part-2-save-as-generic-enrichment-map-output-be"><i class="fa fa-check"></i>PART 2: save as Generic Enrichment Map output (BE)</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#part-3-save-as-generic-enrichment-map-output-ne"><i class="fa fa-check"></i>PART 3: save as Generic Enrichment Map output (NE)</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#part-4-create-an-enrichment-map"><i class="fa fa-check"></i>PART 4: create an enrichment map</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#answers-gprofiler"><i class="fa fa-check"></i>Answers g:Profiler</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#part-5-gsea-run-and-create-an-enrichment-map"><i class="fa fa-check"></i>PART 5: GSEA (run and create an enrichment map)</a><ul>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#answers-gsea"><i class="fa fa-check"></i>Answers GSEA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#part-6-iregulon"><i class="fa fa-check"></i>PART 6: iRegulon</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#dataset-2"><i class="fa fa-check"></i>DATASET 2</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#part-1-reactomefi"><i class="fa fa-check"></i>PART 1: ReactomeFI</a><ul>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#answers-reactome-fi"><i class="fa fa-check"></i>Answers REACTOME FI</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#part-2-genemania"><i class="fa fa-check"></i>PART 2: GeneMANIA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="ass-automation.html"><a href="ass-automation.html"><i class="fa fa-check"></i>Integrated Assignment Bonus - Automation</a><ul>
<li class="chapter" data-level="" data-path="ass-automation.html"><a href="ass-automation.html#goal-of-the-exercise-5"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pathways</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gprofiler_mod3" class="section level1">
<h1>Module 3 Lab: g:profiler Visualization</h1>
<p><strong>This work is licensed under a <a href="http://creativecommons.org/licenses/by-sa/3.0/deed.en_US">Creative Commons Attribution-ShareAlike 3.0 Unported License</a>. This means that you are able to copy, share and modify the work, as long as the result is distributed under the same license.</strong></p>
<p>By Veronique Voisin, Ruth Isserlin, Gary Bader</p>
<div id="goal-of-the-exercise-2" class="section level2">
<h2>Goal of the exercise</h2>
<p><strong>Create an enrichment map and navigate through the network</strong></p>
<p>During this exercise, you will learn how to create an enrichment map from gene-set enrichment results. The enrichment results chosen for this exercise are generated using g:Profiler but an enrichment map can be created directly from output from <a href="http://software.broadinstitute.org/gsea/index.jsp">GSEA</a>, <a href="http://apps.cytoscape.org/apps/bingo">BinGo</a>, <a href="https://amp.pharm.mssm.edu/Enrichr/">Enrichr</a> or alternately from any gene-set tool using the generic enrichment results format.</p>
</div>
<div id="data-2" class="section level2">
<h2>Data</h2>
<p>The data used in this exercise is a list of frequently mutated genes that we used in <a href="gprofiler-lab.html#gprofiler-lab">previous exercise</a>.
Pathway enrichment analysis has been run using g:Profiler and the results have been downloaded as a GEM format.</p>
</div>
<div id="enrichmentmap" class="section level2">
<h2>EnrichmentMap</h2>
<ul>
<li><p>A circle (node) is a gene-set (pathway) enriched in genes that we used as input in g:Profiler (frequently mutated genes).</p></li>
<li><p>edges (lines) represent genes in common between 2 pathways (nodes).</p></li>
<li><p>A cluster of nodes represent overlapping and related pathways and may represent a common biological process.</p></li>
<li><p>Clicking on a node will display the genes included in each pathway.</p></li>
</ul>
</div>
<div id="description-of-this-exercise" class="section level2">
<h2>Description of this exercise</h2>
<p>We run and saved g:Profiler results using different parameters.
An enrichment map represents the result of enrichment analysis as a network where significantly enriched gene-sets that share a lot of genes in common will form identifiable clusters. The visualization of the results as these biological themes will ease the interpretation of the results.</p>
<p>The goal of this exercise is to learn how to:</p>
<ol style="list-style-type: decimal">
<li>upload g:Profiler results into Cytoscape EnrichmentMap to create a map.</li>
<li>upload several g:Profiler results at the same time to create one map and learn how to distinguish the results.</li>
<li>to compare the differences resulting from the use of different g:Profiler parameters at the enrichment map level.</li>
</ol>
</div>
<div id="start-the-exercise-1" class="section level2">
<h2>Start the exercise</h2>
<p>To start the lab practical section, first create a gprofiler_files directoty on your computer and download the files below.</p>
<div class="rmd-datadownload">
<p>
Right click on link below and select “Save Link As…”.
</p>
<p>
Place it in the corresponding module directory of your CBW work directory.
</p>
</div>
<p>Five files are needed for this exercise:</p>
<ol style="list-style-type: decimal">
<li>Enrichment result 1: <a href="./Module3/gprofiler/data/gProfiler_hsapiens_max10000.gem.txt">gProfiler_hsapiens_max10000.gem.txt</a></li>
</ol>
<ul>
<li>In g:Profiler, the parameters that we used to generate this file were:
<ul>
<li>GO_BP no electronic annotation,</li>
<li>Reactome,</li>
<li>Benjamini-HochBerg FDR 0.05</li>
<li>No restriction on the number of genes in a geneset</li>
</ul></li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Enrichment result 2:<a href="./Module3/gprofiler/data/gProfiler_hsapiens_max250.gem.txt">gProfiler_hsapiens_max250.gem.txt</a></li>
</ol>
<ul>
<li>In g:Profiler, the parameters that we used were:
<ul>
<li>GO_BP no electronic annotation,</li>
<li>Reactome,</li>
<li>Benjamini-HochBerg FDR 0.05.</li>
<li>The results were filtered using the g:Profiler Term size slidebar and only the enriched gene-sets that contain equal or less than 250 genes per gene-set were included in the result file (gProfiler_hsapiens_max250.gem.txt).</li>
</ul></li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>Enrichment result 3:<a href="./Module3/gprofiler/data/gProfiler_hsapiens_Baderlab_max250_gem.txt">gProfiler_hsapiens_Baderlab_max250_gem.txt</a></li>
<li>Pathway database 1 (.gmt):<a href="./Module3/gprofiler/data/gprofiler_full_hsapiens.name.gmt">gprofiler_full_hsapiens.name.gmt</a></li>
</ol>
<ul>
<li>This file can be downloaded directly or can be been created by concatenating the hsapiens.GO/BP.name.gm and the hsapiens.REAC.name.gmt files contained in the g:Profiler gprofiler_hsapiens.name folder.</li>
</ul>
<ol start="5" style="list-style-type: decimal">
<li>Pathway database 2 (.gmt):<a href="./Module3/gprofiler/data/Human_GOBP_AllPathways_no_GO_iea_June_20_2019_symbol_max250gssize.gmt">Human_GOBP_AllPathways_no_GO_iea_June_20_2019_symbol_max250gssize.gmt</a></li>
</ol>
</div>
<div id="exercise-1a" class="section level2">
<h2>Exercise 1a</h2>
<div id="step-1" class="section level3">
<h3>Step 1</h3>
<p>Launch Cytoscape and open the EnrichmentMap App</p>
<p>1a. Double click on Cytoscape icon</p>
<p>1b. Open EnrichmentMap App</p>
<ul>
<li><p>In the Cytoscape top menu bar:</p></li>
<li><p>Click on Apps -&gt; EnrichmentMap</p></li>
</ul>
<p><img src="Module3/gprofiler/images/EM1.png"  /></p>
<ul>
<li>A ‘Create Enrichment Map’ window is now opened.</li>
</ul>
</div>
<div id="step-2" class="section level3">
<h3>Step 2</h3>
<p>Create an enrichment map from 2 datasets and with a gmt file.</p>
<p>2a. In the ‘Create Enrichment Map’ window, drag and drop the 2 enrichment files gProfiler_hsapiens_max10000.gem.txt and
gProfiler_hsapiens_max250.gem.txt.</p>
<p><img src="Module3/gprofiler/images/gem0.png" alt="workflow" width="100%" /></p>
<p>2b. In the white box, click on "gProfiler_hsapiens_max250.gem (Generic/gProfiler)</p>
<p>2c. On the right side, go to the <em>GMT</em> field, click on the 3 radio button (…) and locate the file gprofiler_full_hsapiens.name.gmt that you have saved on your computer to upload it.</p>
<p><img src="Module3/gprofiler/images/gem1.png" alt="workflow" width="100%" /></p>
<p>2d. In the white box, click on "gProfiler_hsapiens_max10000 (Generic/gProfiler)</p>
<p>2e. On the right side, go to the <em>GMT</em> field, click on the 3 radio button (…) and locate the file gprofiler_full_hsapiens.name.gmt that you have saved on your computer to upload it.</p>
<p>2f. Locate the <em>FDR q-value cutoff</em> field and set the value to 0.001</p>
<p>2g. Select the <em>Connectivity</em> slide bar to <em>sparse</em>.</p>
<p><img src="Module3/gprofiler/images/gem2.png" alt="workflow" width="100%" /></p>
2h. Click on <em>Build</em>.
* a status bar should pop up showing progress of the Enrichment map build.
<p align="center">
<img src="Module3/gprofiler/images/gem3.png" alt="workflow" width="60%"/>
</p>
</div>
<div id="step3-explore-the-results" class="section level3">
<h3>Step3: Explore the results:</h3>
<p>In the EnrichmentMap control panel located at the left:</p>
<ul>
<li>select the 2 Data Sets (checked by default)</li>
<li>set Chart Data o <em>Color by Data Set</em></li>
<li>select <em>Publication Ready</em> to remove gene-set label to have a global view of the map.</li>
</ul>
<div class="rmd-tip">
<p>
un-select <em>Publication Ready</em> when you explore the map in more detail to see the gene-set names.
</p>
</div>
<p align="center">
<img src="Module3/gprofiler/images/gem3a.png" alt="workflow" width="70%" />
</p>
<p>On the map, a node that is coloured both green and blue is a gene-set that is found in the both of the 2 gProfiler result sets that we have been uploaded.
* A node that is blue is a gene-set that is found only in the file <em>gProfiler_hsapiens_max10000</em> .
* A blue edge represents genes that overlap between gene-sets found in the file <em>gProfiler_hsapiens_max10000</em>.
* A green edge represents genes that overlap between gene-sets found in the file <em>gProfiler_hsapiens_max250.gem</em>.</p>
<p><img src="Module3/gprofiler/images/gem6.png" alt="workflow" width="100%" /></p>
<p>We can see clusters of blue nodes. All these nodes contain gene-sets that have more than 250 genes. Explore the detailed view (see below) to see if this cluster corresponds to informative terms.</p>
<div class="rmd-question">
<p>
Would you have lost information by filtering gene-sets larger than 250 genes?
</p>
</div>
</div>
<div id="explore-detailed-results" class="section level3">
<h3>Explore Detailed results</h3>
<ul>
<li>In the Cytoscape menu bar, select ‘View" and ’Show Graphic Details’ to display node labels.</li>
</ul>
<div class="rmd-caution">
<p>
Make sure you have unselected “Publication Ready” in the EnrichmentMap control panel.
</p>
</div>
<ul>
<li><p>Zoom in to be able to read the labels and navigate the network using the bird eye view (blue rectangle).</p></li>
<li><p>Select a node and visualize the <em>Table Panel</em></p>
<ul>
<li><p>Click on a node</p></li>
<li><p>For this example the node <em>“Signaling by Notch”</em> has been selected.</p></li>
</ul></li>
</ul>
<div class="rmd-tip">
<p>
you can type it in the search bar, quotes are important.
</p>
</div>
<p><img src="Module3/gprofiler/images/gem8.png" alt="workflow" width="100%" /></p>
<p>When the node is selected, it is highlighted in <font color="yellow">yellow</font>.
In table panel, we can see the genes included in the gene-set. A green colored box indicates that the gene is in the gene-set(pathway) and in our gene list. A gray colored box indicated that the gene is in the gene-set but not in our gene list.</p>
</div>
</div>
<div id="exercise-1b" class="section level2">
<h2>Exercise 1b</h2>
<p>Create an enrichment map without a gmt file to compare the results with Exercise 1a.</p>
<ul>
<li>Go to Control Panel and select the EnrichmentMap tab.</li>
<li>Click on the “+” sign to re-open the <em>Create Enrichment Map</em> window.
<p align="center">
<img src="Module3/gprofiler/images/gem7.png" alt="workflow" width="50%" />
</p></li>
<li>In the white box, select the "gProfiler_hsapiens_max250.gem (Generic/gProfiler) file</li>
<li>Locate the GMT field and delete the file name , leaving it blank.</li>
<li>In the white box, select the "gProfiler_hsapiens_max10000 (Generic/gProfiler) file</li>
<li>Locate the GMT field and delete the file name , leaving it blank.</li>
<li>Use same parameters as in <a href="gprofiler-mod3.html#exercise-1a">exercise 1a</a>: FDR q-value cutoff of 0.001 and Connectivity to sparse.</li>
<li>Click on <em>Build</em></li>
</ul>
<p><img src="Module3/gprofiler/images/gem5.png" alt="workflow" width="100%" /></p>
<p>Explore the results:</p>
<p>In the EnrichmentMap control panel located at the left:</p>
<ul>
<li>select the 2 Data Sets (selecteded by default)</li>
<li>set Chart Data o <em>Color by Data Set</em></li>
<li>select <em>Publication Ready</em> to remove gene-set label to have a global view of the map.</li>
</ul>
<div class="rmd-tip">
<p>
uncheck this box when you explore the map in details to see the gene-set names.
</p>
</div>
<p align="center">
<img src="Module3/gprofiler/images/gem3a.png" alt="workflow" width="70%" />
</p>
<p>On the map, a node that is coloured both green and blue is a gene-set that is found in the both of the 2 gProfiler result sets that we have been uploaded.</p>
<ul>
<li>A node that is blue is a gene-set that is found only in the file <em>gProfiler_hsapiens_max10000</em> .</li>
<li>A blue edge represents genes that overlap between gene-sets found in the file <em>gProfiler_hsapiens_max10000</em>.</li>
<li>A green edge represents genes that overlap between gene-sets found in the file gProfiler_hsapiens_max250.gem.</li>
</ul>
<p><img src="Module3/gprofiler/images/gem4.png" alt="workflow" width="100%" /></p>
<p><strong>Conclusion of exercises 1 a and 1b:</strong></p>
<p>Loading a gmt file to create an enrichment map from g:Profiler result is optional. However, there are 2 main beneficial aspects to uploading a gmt file:</p>
<ol style="list-style-type: decimal">
<li>the map will be less condensed and easier to read and interpret.<br />
</li>
<li>clicking on a node will display all genes in the gene-set and not only genes included in our query list.</li>
</ol>
</div>
<div id="exercise-1c" class="section level2">
<h2>Exercise 1c</h2>
<p>Create an enrichment map from the results of g:Profiler generated using the custom Baderlab gene-set file.
To get a map that is easy to read and that does not display too many gene-sets, one option is to focus the analysis on gene-sets (pathways) that contain 250 genes or less. We prefiltered our pathway database prior to upload it into g:Profiler so that FDR is calculated only on these gene-sets ( as opposed to exercise 1a where the FDR was calculated on all gene-sets and then some gene-sets &gt; 250 genes were excluded from the result file ).</p>
<ul>
<li><p>filtered gmt file: (<a href="./Module3/gprofiler/data/Human_GOBP_AllPathways_no_GO_iea_June_24_2019_symbol_max250gssize.gmt">Human_GOBP_AllPathways_no_GO_iea_June_24_2019_symbol_max250gssize.gmt</a>).</p></li>
<li><p>We have uploaded this file as a custom gmt file in g:Profiler and run the query.</p></li>
<li><p>To create an enrichment map of these results:</p></li>
<li><p>Go to Control Panel and select the EnrichmentMap tab.</p></li>
<li><p>Click on the “+” sign to re-open the <em>Create Enrichment Map</em> window.</p>
<p align="center">
<p><img src="Module3/gprofiler/images/gem7.png" alt="workflow" width="50%" /></p>
</p></li>
<li><p>In the white box, select the "gProfiler_hsapiens_Baderlab_max250_gem.txt (Generic/gProfiler) file</p></li>
<li><p>Locate the GMT field and upload the file “Human_GOBP_AllPathways_no_GO_iea_June_24_2019_symbol_max250gssize.gmt”.</p></li>
</ul>
<p><img src="Module3/gprofiler/images/gem9.png" alt="workflow" width="100%" /></p>
<p>Explore the results:</p>
<p><img src="Module3/gprofiler/images/gem10.png" alt="workflow" width="100%" /></p>
<div class="rmd-caution">
<p>
SAVE YOUR CYTOSCAPE SESSION (.cys) FILE !
</p>
</div>

<p>#Module 3 Lab: GSEA Visualization {#gsea_mod3}</p>
<p><strong>This work is licensed under a <a href="http://creativecommons.org/licenses/by-sa/3.0/deed.en_US">Creative Commons Attribution-ShareAlike 3.0 Unported License</a>. This means that you are able to copy, share and modify the work, as long as the result is distributed under the same license.</strong></p>
<p><em><font color="#827e9c">By Veronique Voisin, Ruth Isserlin, Gary Bader</font></em></p>
</div>
<div id="goal-of-the-exercise-3" class="section level2">
<h2>Goal of the exercise:</h2>
<p><strong>Create an enrichment map and navigate through the network</strong></p>
<p>During this exercise, you will learn how to create an EnrichmentMap from gene-set enrichment results. The enrichment tool chosen for this exercise is GSEA but an enrichment map can be created from various gene-set tools using the generic format or the more specific <a href="https://biit.cs.ut.ee/gprofiler/gost">g:Profiler</a>,<a href="https://amp.pharm.mssm.edu/Enrichr/">Enrichr</a>, <a href="https://david.ncifcrf.gov/">DAVID</a> or <a href="http://apps.cytoscape.org/apps/bingo">BinGo</a> formats.</p>
<p><strong>Post analysis (add drug target gene-sets to the network)</strong></p>
<p>As second part of the exercise, you will learn how to expand the network by adding an extra layer of information.</p>
<p><strong>Autoannotate</strong></p>
<p>A last optional exercise guides you toward the creation of automatically generated cluster labels to the network.</p>
</div>
<div id="data-3" class="section level2">
<h2>Data</h2>
<p>The data used in this exercise is gene expression data obtained from high throughput RNA sequencing.
They correspond to Ovarian serous cystadenocarcinoma samples. We use precomputed results of the GSEA analysis <a href="gsea-lab.html#gsea-lab">Module 2 lab - gsea</a> to create an enrichment map with the aim to transform the tabular format in a network to better visualize the relationships between the significant gene-sets:</p>
<p><img src="Module3/gsea/images/workflow.png" alt="workflow" width="750" /></p>
</div>
<div id="gsea-output-and-enrichmentmap" class="section level2">
<h2>GSEA output and EnrichmentMap</h2>
<p>The output results of GSEA are 2 tables:</p>
<ul>
<li><p>One table (<em>pos</em>) contains all pathways with an enrichment score (significant or not) related to enrichment of the mesenchymal category (positive score).</p></li>
<li><p>One table (<em>neg</em>) contains all pathways with an enrichment score (significant or not) related to enrichment of the immunoreactive category (negative score).</p></li>
<li><p>These 2 tables are uploaded using the EnrichmentMap App which will create a network of mesenchymal and immunoreactive pathways that have a significant score (FDR &lt;= 0.0001) for clearer visualization of the results.</p></li>
</ul>
<div id="enrichmentmap-1" class="section level3">
<h3>EnrichmentMap</h3>
<ul>
<li><p>A <font color="red">red circle (node)</font> is a pathway specific of the mesenchymal type.</p></li>
<li><p>A <font color="blue">blue circle (node)</font> is a pathway specific of the immunoreactive type.</p></li>
<li><p><font color="green">Green edges</font> represent genes in common between 2 pathways (nodes).</p></li>
<li><p>A cluster of nodes represent overlapping and related pathways and may represent a common biological process.</p></li>
<li><p>Clicking on a node will display the genes included in each pathway.</p></li>
</ul>
</div>
</div>
<div id="start-the-exercise-2" class="section level2">
<h2>Start the exercise</h2>
<p>To start the lab practical section, first download the files.</p>
<div class="rmd-datadownload">
<p>
Right click on link below and select “Save Link As…”.
</p>
<p>
Place it in the corresponding module directory of your CBW work directory.
</p>
</div>
<p>7 Files are needed to create the enrichment map for this exercise (please download these files on your computer):</p>
<ol style="list-style-type: decimal">
<li><p>GMT (file containing all pathways and corresponding genes) - <a href="./Module3/gsea/data/Human_GOBP_AllPathways_no_GO_iea_July_01_2020_symbol.gmt">Human_GOBP_AllPathways_no_GO_iea_July_01_2020_symbol.gmt</a></p></li>
<li><p>Expression (file containing the RNAseq data for all samples and all genes) - <a href="./Module3/gsea/data/MesenchymalvsImmunoreactive_rnaseq_expression.txt">MesenchymalvsImmunoreactive_rnaseq_expression.txt</a></p></li>
<li><p>Enrichments 1 (GSEA results for the “pos” mesenchymal phenotype) - <a href="./Module3/gsea/data/gsea_report_for_na_pos_1595281162211.xls">gsea_report_for_na_pos_1595281162211.xls</a></p></li>
<li><p>Enrichments 1 (GSEA results for the “neg” immunoreactive phenotype) - <a href="./Module3/gsea/data/gsea_report_for_na_neg_1595281162211.xls">gsea_report_for_na_neg_1595281162211.xls</a></p></li>
<li><p>Rank file (file that has been used as input to GSEA) - <a href="./Module3/gsea/data/MesenchymalvsImmunoreactive_edger_ranks.rnk">MesenchymalvsImmunoreactive_edger_ranks.rnk</a></p></li>
<li><p>Classes (define which samples are mesenchymal and which samples are immunoreactive) - <a href="./Module3/gsea/data/MesenchymalvsImmunoreactive_rnaseq_classes.cls">MesenchymalvsImmunoreactive_rnaseq_classes.cls</a></p></li>
<li><p>Drug target database (preselection of 4 drugs and their target genes u sed in the poast -analysis exercise, <a href="http://www.drugbank.ca/" class="uri">http://www.drugbank.ca/</a>) - <a href="./Module3/gsea/data/Human_DrugBank_approved_symbol_selected.gmt">Human_DrugBank_approved_symbol_selected.gmt</a></p></li>
</ol>
<p>Follow the steps described below at your own pace:</p>
<div id="step-1-1" class="section level3">
<h3>Step 1</h3>
<p>Launch Cytoscape and open EnrichmentMap App</p>
<p><strong>1a</strong>. Double click on the Cytoscape icon</p>
<p><strong>1b</strong>. Open EnrichmentMap App</p>
<ul>
<li><p>In the top menu bar:</p>
<ul>
<li>Click on Apps -&gt; EnrichmentMap</li>
</ul></li>
</ul>
<p><img src="Module3/gsea/images/EM1.png"  /></p>
<p>A ’Create EnrichmentMap window is now opened.</p>
</div>
<div id="step-2-1" class="section level3">
<h3>Step 2</h3>
<p>Create an enrichment map</p>
<p><strong>2a</strong>. In the ‘Create EnrichmentMap’ window, add a dataset of the GSEA type by clicking on the ‘+’ plus sign.</p>
<p><img src="Module3/gsea/images/EMinput.png" width="750" /></p>
<p><strong>2b</strong>. Specify the following parameters and upload the specified files:</p>
<ul>
<li><p><em>Name</em>: leave default or a name of your choice like “GSEAmap”</p></li>
<li><p><em>Analysis Type</em>: GSEA</p></li>
<li><p><em>Enrichments Pos</em>: gsea_report_for_na_pos_1595281162211.xls</p></li>
<li><p><em>Enrichments Ned</em>: gsea_report_for_na_neg_1595281162211.xls</p></li>
<li><p><em>GMT</em> : Human_GOBP_AllPathways_no_GO_iea_July_01_2020_symbol.gmt</p></li>
<li><p><em>Ranks</em>: MesenchymalvsImmunoreactive_edger_ranks.rnk</p></li>
<li><p><em>Expressions</em> : MesenchymalvsImmunoreactive_rnaseq_expression.txt.</p>
<div class="rmd-tip">
<p>
this field is optional but recommended
</p>
</div></li>
<li><p><em>Classes</em>: MesenchymalvsImmunoreactive_rnaseq_classes.cls</p>
<div class="rmd-tip">
<p>
this field is optional.
</p>
</div></li>
<li><p><em>Phenotypes</em>: In the text boxes place <em>Mesenchymal</em> as the Positive phenotype <em>Immunoreactive</em> as the Negative phenotype. Mesenchymal will be associated with red nodes because it corresponds to the positive phenotype and Immunoreactive will be associated with the blue nodes because it corresponds to the negative phenotype.</p></li>
<li><p>set FDR q-value cutoff to 0.0001 (= only gene-sets significantly enriched at a value of 0.0001 or less will be displayed on the map).</p>
<div class="rmd-tip">
<p>
it will be displayed as 1E-04 in the scientific notation.
</p>
</div></li>
</ul>
<p><strong>2c</strong>. Click on <em>Build</em></p>
<p><img src="Module3/gsea/images/EMinput2.png" alt="EM" width="750" /></p>
<div class="rmd-caution">
<p>
We populated the fields manually. If you work with your own data, a way to populate automatically the fields is to drag and drop your GSEA folder in the ‘Data Set’ window. You are encouraged to give it a try once you finished the lab with your own GSEA results.
</p>
</div>
<p><strong>Unformatted results</strong>:</p>
<div class="rmd-note">
<p>
layout will be different for each user (there is a random seed in the layout algorithm) but it does not change the results or interpretation (the connections are the same, only the display is different).
</p>
</div>
<p><img src="Module3/gsea/images/EM3.png" alt="EM" width="750" /></p>
</div>
<div id="step-3" class="section level3">
<h3>Step 3</h3>
<p>Navigate the enrichment map to gain a better understanding of a EnrichmentMap network.</p>
<p>General layout of Cytoscape panel: In addition to the main window where the network is displayed, there are 2 panels: the Control Panel on the left side and the Table Panel at the bottom of the window.</p>
<p>Steps:</p>
<p><strong>3a</strong>. In the Cytocape menu bar, select <em>View</em> and <em>Show Graphic details</em>. It will turn the squared nodes into circles and the gene-set labels will be visible.</p>
<p><img src="Module3/gsea/images/EM5.png" alt="EM" width="350" /></p>
<p><strong>3b</strong>: Zoom in or out using + or - in toolbar or scroll button on mouse until you are able to read the labels comfortably.</p>
<p><img src="Module3/gsea/images/EM6.png" alt="EM" width="350" /></p>
<p><strong>3c</strong>: Use the bird’s eye view (located at the bottom of the control panel) to navigate around the network by moving the blue rectangle using the mouse or trackpad.</p>
<p><img src="Module3/gsea/images/EM7.png" alt="EM" width="350" /></p>
<p><strong>3d</strong>: Click on an individual node of interest.</p>
<p>For this example, you could use <em>Integrin signalling pathway</em>.</p>
<div class="rmd-tip">
<p>
If you are unable to locate <em>Integrin signalling pathway</em>, type “Integrin signalling pathway” in the search box (quotes are important). Selected nodes appear yellow (or highlighted) in the network.
</p>
</div>
<p><strong>3e</strong>. In the Table Panel in the <em>EM Heat map</em> tab change:</p>
<ul>
<li><p>Expressions: Row Norm*</p></li>
<li><p>Compress: -None-</p></li>
</ul>
<p><img src="Module3/gsea/images/EM8.png" alt="EM" width="750" /></p>
<div class="rmd-tip">
<p>
Genes in the heatmap that are highlighted yellowm (rank column) represent genes part of the leading edge for this gene set, i.e. contributed the most to the enriched phenotype. Leading edge genes will only be highlighted if an individual node has been selected and the Enrichment Map was created from GSEA results. Troubleshooting: if you don’t see the sort column highlighted in yellow, reselect the node of interest and click on the GSEARanking Data Set 1 text in the EM Heatmap tab.
</p>
</div>
</div>
<div id="step-4" class="section level3">
<h3>Step 4</h3>
<p>Use Filters to automatically select nodes on the map: Move the blue nodes to the left side of the window and the red nodes to the right side of the window.</p>
<p><strong>4a</strong>. Locate the <em>Filter</em> tab on the side bar of the <em>Control Panel</em>.</p>
<p><strong>4b</strong>. Click on the + sign to view the menu and select <em>Column Filter</em>.</p>
<p><strong>4c</strong>. From the <em>Choose column …</em> box, select <em>Node: ES_dataset1</em> and set filter values from -0.953 and 0 inclusive.</p>
<p><strong>4d</strong>. The blue nodes are now automatically selected. Zoom out to be able to look at the entire network and drag all blue nodes to the left side of the screen.</p>
<p><img src="Module3/gsea/images/EM9.png" alt="EM" width="750" /></p>
<p><strong>4e</strong>. Optional. Change <em>is</em> to <em>is not</em> to select the red nodes.</p>
<p><img src="Module3/gsea/images/EM10.png" alt="EM" width="750" /></p>
<p>Note: the red pathways (nodes) are specific of the mesenchymal phenotype. There were listed in the <em>pos</em> table of the GSEA results. The enrichment score (ES) in this table are all positive values. The blue pathways are specific of the immunoreactive phenotype and were listed in the <em>neg</em> table of the GSEA results. The ES in this table are all negative values. This information has been used to set the filtering criteria.</p>
</div>
<div id="step-5" class="section level3">
<h3>Step 5</h3>
<p>Add drug target gene-sets to the network (Add Signature Gene-Sets…).</p>
<p><strong>5a</strong>. In Control Panel, go to the EnrichmentMap tab and click on “Add Signature Gene Sets…” located below the white ‘Data Sets:’ window. A window named "EnrichmentMap: Add Signature Gene Sets (Post-Analysis) is now opened.</p>
<p><img src="Module3/gsea/images/drug1.png" alt="EM" width="250" /></p>
<p><strong>5b</strong>. Using the ‘Load from File…’ button, select the <em>Human_DrugBank_approved_symbol_selected.gmt</em> file that you saved on your computer.</p>
<p><img src="Module3/gsea/images/drug2.png" alt="EM" width="750" /></p>
<p><strong>5c</strong>. Click on “Finish”.</p>
<p>Note: Four additional nodes are now added to the network and visible as grey diamonds. Dotted orange edges represent their overlap with the nodes of our network. These additional nodes represent gene targets of some approved drugs and these genes are either specific of the mesenchymal type (dotted orange edges connected to red nodes) or specific of the immunoreactive type (dotted orange edges connected to blue nodes).</p>
<p><img src="Module3/gsea/images/drug3.png" alt="EM" width="750" /></p>
<p><img src="Module3/gsea/images/drug4.png" alt="EM" width="750" /></p>
<div class="rmd-tip">
<p>
more info using this link: <a href="https://enrichmentmap.readthedocs.io/en/latest/PostAnalysis.html" class="uri">https://enrichmentmap.readthedocs.io/en/latest/PostAnalysis.html</a>
</p>
</div>
</div>
<div id="step-6" class="section level3">
<h3>Step 6</h3>
<p>Auto-annotate the network with cluster labels.</p>
<div class="rmd-note">
<p>
The Apps WordCloud, ClusterMaker and Autoannotate have to be installed.
</p>
</div>
<div class="rmd-note">
<p>
if you run step 5, delete the drug targets diamond nodes and associated edge before performing step 6: select the 4 nodes and associated dotted orange edges by browsing the mouse and click “delete” on your keyboard or in the Cytoscape menu, ‘Edit’, ‘Delete Selected Nodes and Edges’.
</p>
</div>
<p><strong>6a</strong>. In the menu bar, select <em>Apps</em>, <em>AutoAnnotate</em>, <em>New Annotation Set …</em>. An <em>Autoannotate: Create Annotation Set</em> window pops up. Leave default settings and click on <em>Create Annotations</em>.</p>
<p><img src="Module3/gsea/images/EM12.png" alt="EM" width="350" /></p>
<p>In the case that an <em>AutoAnnotate</em> window pops up with the message <em>AutoAnnotate will manage all groups in this network view</em>, click <em>OK</em>.</p>
<p><img src="Module3/gsea/images/EM13.png" alt="EM" width="350" /></p>
<p><img src="Module3/gsea/images/EM14.png" alt="EM" width="750" /></p>
<div class="rmd-note">
<p>
The network is now subdivided into clusters that are represented by ellipses. Each of these clusters are composed of pathways (nodes) interconnected by many common genes. These pathways represent similar biological processes. The app WordCloud take all the labels of the pathways in one cluster and summarize them as a unique cluster label displayed at the top of each ellipse.
</p>
</div>
<div class="rmd-tip">
<p>
<strong>Tip 1</strong>: further editing and formatting can be performed on the AutoAnnote results using the <em>AutoAnnotate Display</em> in the <em>Results Panels</em> located at the right side of the window. For example, it is possible to change Ellipse to Rectangle, uncheck <em>Scale font by cluster size</em> and increase the <em>Font Scale</em> using the scaling bar. It is also possible to reduce the length of the cluster label by checking the “Word Wrap” option.
</p>
<p>
<strong>Tip 2</strong>: The AutoAnnotate window on the left side in Result Panel contains the list of all clusters. Clicking on a cluster label will highlight in yellow all nodes in this cluster. It is then easy to move the nodes using the mouse to avoid cluster overlaps.
</p>
</div>
<p><img src="Module3/gsea/images/AAStep6dnew.png" alt="workflow" width="750" /></p>
<p><strong>6b</strong>. If your network is dense and clusters are overlapping, there is an option in AutoAnnotate that you can try that expands the network to prevent cluster overlap. In the menu bar, select <em>Apps</em>, <em>AutoAnnotate</em>, <em>New Annotation Set …</em>. An <em>Autoannotate: Create Annotation Set</em> window pops up. Check the <em>Layout network to prevent cluster overlap</em> and click on <em>Create Annotations</em>.</p>
<p><img src="Module3/gsea/images/AA6f.png" alt="workflow" width="500" /></p>
<p><img src="Module3/gsea/images/AA6g.png" alt="workflow" width="750" /></p>
</div>
<div id="step-7" class="section level3">
<h3>Step 7</h3>
<p>Visualize genes in a pathway/node of interest using the apps STRING and GeneMANIA. This will create a protein-protein interaction network using the genes included in the pathway. Note: GeneMANIA theory and practical will be covered in next module.</p>
<p><strong>7a</strong>: Click on an individual node of interest.</p>
<p>For this example, you could use <em>Integrins in angiogenesis</em>.</p>
<div class="rmd-tip">
<p>
If you are unable to locate <em>Integrins in angiogenesis</em>, type “Integrins in angiogenesis” in the search box (quotes are important). The selected node appears yellow (or highlighted) in the network. If you have annotated your network, it should be included in the <em>ecm receptor interaction</em> cluster.
</p>
</div>
<p><strong>7b</strong>: Right Click on the node of interest to diplay the option menu. Select <em>Apps</em>, <em>EnrichmentMap - Show in STRING</em>
A <em>STRING Protein Query</em> box appears. Unselect all genes <em>Select None</em> and select only the collagen genes with gene name starting with <em>COL</em>. Click on <em>OK</em>.</p>
<p><img src="Module3/gsea/images/Stringa.png" alt="workflow" width="500" /></p>
<p><img src="Module3/gsea/images/Stringb.png" alt="workflow" width="500" /></p>
<p><img src="Module3/gsea/images/Stringc.png" alt="workflow" width="750" /></p>
<p><strong>7c</strong>:Go back to enrichment map network. In Control Panel (left side of the window), select the “Network” tab and click on the Enrichment Map network as shown in below screenshot.</p>
<p><img src="Module3/gsea/images/sting_genemani.png" alt="workflow" width="500" /></p>
<p><strong>7d</strong>: Search again for the node labelled <em>Integrins in angiogenesis</em> as in Step 7a. Right Click on the node of interest to diplay the option menu. Select <em>Apps</em>, <em>EnrichmentMap - Show in GeneMANIA</em>. A <em>GeneMANIA Query</em> box appears. Unselect all genes <em>Select None</em> and select only the collagen genes with gene name starting with <em>COL</em>. Click on <em>OK</em>.</p>
<p><img src="Module3/gsea/images/Stringa.png" alt="workflow" width="500" /></p>
<p><img src="Module3/gsea/images/Genemaniab.png" alt="workflow" width="500" /></p>
<p><img src="Module3/gsea/images/genemaniaquery.png" alt="workflow" width="500" /></p>
<p><img src="Module3/gsea/images/Genemaniac.png" alt="workflow" width="750" /></p>
<div class="rmd-tip">
<p>
It is possible to view gene expression data for the nodes in the STRING network. See the section <a href="https://enrichmentmap.readthedocs.io/en/latest/Integration.html" class="uri">https://enrichmentmap.readthedocs.io/en/latest/Integration.html</a> and try it out after the workshop.
</p>
</div>
<div class="rmd-caution">
<p>
SAVE YOUR FILE!
</p>
</div>

<p>#Module 3 Lab: (Bonus) Automation {#automation}</p>
<p><strong>This work is licensed under a <a href="http://creativecommons.org/licenses/by-sa/3.0/deed.en_US">Creative Commons Attribution-ShareAlike 3.0 Unported License</a>. This means that you are able to copy, share and modify the work, as long as the result is distributed under the same license.</strong></p>
<p><em><font color="#827e9c">By Ruth Isserlin</font></em></p>
</div>
</div>
<div id="goal-of-the-exercise-4" class="section level2">
<h2>Goal of the exercise:</h2>
<p><strong>Run an enrichment analysis and Create an enrichment map automatically from R/Rstudio</strong></p>
<p>During this exercise, you will apply what you have learnt in Module 2 labs and Module 3 labs but instead of performing them manually you will automate the process using R/Rstudio. We will use all the same data and programs we used in the previous labs but we will control them from R.</p>
<div id="set-up" class="section level3">
<h3>Set Up</h3>
<p>Before starting this exercise,</p>
<ol style="list-style-type: decimal">
<li>Install R and RStudio (if not already installed)</li>
</ol>
<ul>
<li>For these exercises you also require Rstudio. If you do not already have RStudio installed please follow the below instructions to install it.</li>
</ul>
<ol style="list-style-type: lower-alpha">
<li>Install R.</li>
</ol>
<ul>
<li>Go to: <a href="https://cran.rstudio.com/" class="uri">https://cran.rstudio.com/</a></li>
</ul>
<p><img src="Module3/automation/images/downloadR.png" alt="Load data" /></p>
<ul>
<li>If installing on Windows select “install R for the first time” to get to the required package.</li>
</ul>
<p><img src="Module3/automation/images/downloadR_win.png" alt="Load data" /></p>
<ul>
<li>Download and install</li>
</ul>
<p><img src="Module3/automation/images/downloadR_win2.png" alt="Load data" /></p>
<ol start="2" style="list-style-type: lower-alpha">
<li>Install R Studio.</li>
</ol>
<pre><code>* Go To https://rstudio.com/products/rstudio/download/#download
* Install RStudio for your platform</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Download example R notebooks from <a href="https://github.com/BaderLab/Cytoscape_workflows" class="uri">https://github.com/BaderLab/Cytoscape_workflows</a>.</li>
</ol>
<ul>
<li><p>This repository contains example R Notebooks that automate the enrichment map pipeline.<br />
</p></li>
<li><p>There are two ways you can download this collection:</p>
<ol style="list-style-type: lower-alpha">
<li>If you are familiar with git then we recommend you fork the repo and use it like you would use any github repo.</li>
</ol>
<p><img src="Module3/automation/images/git_fork.png" alt="Load data" /></p>
<ol start="2" style="list-style-type: lower-alpha">
<li>download the collection as a zip file - unzip folder and place in CBW working directory</li>
</ol>
<p><img src="Module3/automation/images/git_download.png" alt="Load data" /></p></li>
</ul>
<div class="rmd-tip">
<p>
If you are new to git and want to learn more about code versioning then we recommend you read the following <a href="https://guides.github.com/introduction/git-handbook/">tutorial</a> And check out <a href="https://desktop.github.com/">Github Desktop</a> - a desktop application to communicate with github.
</p>
</div>
</div>
<div id="step-1---launch-rstudio" class="section level3">
<h3>Step 1 - launch RStudio</h3>
<ul>
<li>Launch RStudio by double clicking on the installed program icon.</li>
</ul>
</div>
<div id="step-2---create-a-new-project" class="section level3">
<h3>Step 2 - create a new project</h3>
<ul>
<li>Create a new project - File -&gt; New R Project …</li>
</ul>
<p><img src="Module3/automation/images/Rproject_new.png" alt="new project" width="50%"/></p>
<ul>
<li>Select Create project from - “Existing Directory”</li>
</ul>
<p><img src="Module3/automation/images/Rproject_existing_dir.png" alt="existing dir" width="50%"/></p>
<ul>
<li>Click on the Browse button</li>
</ul>
<p><img src="Module3/automation/images/Rproject_browse.png" alt="browse" width="50%" /></p>
<ul>
<li>Navigate to the EnrichmentMapPipeline directory that is found in the directory you downloaded and unzipped from github. (for example, if it is still in your downloads directory go to ~/Downloads/Cytoscape_workflows/EnrichmentMapPipelines)</li>
</ul>
<p><img src="Module3/automation/images/Rproject_open_proj.png" alt="open project" width="50%"/></p>
</div>
<div id="step-3---open-example-up-rnotebook" class="section level3">
<h3>Step 3 - Open example up RNotebook</h3>
<ul>
<li><p>Open the RNotebook <strong>Protocol2_createEM.Rmd</strong></p>
<ul>
<li><p>Go to File –&gt; Open File …</p>
<p><img src="Module3/automation/images/Rproject_openfile.png" alt="open project" width="50%"/></p></li>
<li><p>Click on <strong>Protocol2_createEM.Rmd</strong></p></li>
</ul></li>
</ul>
<div class="rmd-tip">
<p>
If the file is not found in the first directory that RStudio opens up then go back and make sure that you created an Rproject from an “Existing directory” in the previous step.
</p>
</div>
</div>
<div id="step-4---define-notebook-parameters" class="section level3">
<h3>Step 4 - Define Notebook parameters</h3>
<div class="rmd-tip">
<p>
Setting up Notebooks with parameters allows you to re-run the same notebook with different datasets very easily. Whereever possible create notebooks with parameters so you can re-use them
</p>
</div>
<p><img src="Module3/automation/images/Rproject_notebook_params.png" alt="open project"/></p>
<p>Descriptions of each of the parameters:</p>
<ol style="list-style-type: decimal">
<li><strong>analysis_name</strong> - change this field to whatever you want your analysis to be called. GSEA directories generated will be named using this string. It will have additional characters added to it as GSEA generates a random number that it associates with each of its output directories.</li>
<li><strong>working_dir</strong> - path to directory containing the data files we will be analyzing.</li>
<li><strong>rnk_file</strong> - NOT Optional. This notebook runs GSEA preranked and this is the only file that is required. for details on the specifications of this file see <a href="#gsea_lab">Module 3 - gsea lab</a>.</li>
<li><strong>class_file</strong> - (Optional) this file is a GSEA specific data file and is used for better visualization in the heat map viewer of enrichment map. For details of the creation of this file see - <a href="https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#Phenotype_Data_Formats">GSEA documentation - class files</a></li>
<li><strong>expression_file</strong> - (Optional) this file contains the expression values for each of the experiments used in to create your rank file. It is used for better visualization in the heat map viewer of enrichment map.</li>
<li><strong>run_gsea</strong> - set to yes or no. This variable specifies whether the notebook should run GSEA. GSEA can take a while to run so if you have already performed the GSEA analysis and simply want to create an enrichment map you can set this variable to “no”. <strong>If this variable is set to no then you need to specify the path and the name of the GSEA directory in “gsea_directory” parameter.</strong></li>
<li><strong>gsea_jar</strong> - full path to the gsea jar. In the latest version of GSEA this is actually a bat(for windows users) or a sh (for moc and linux users) script. If using GSEA 3.0 or older set this variable to the full path to the gsea.jar <strong>We do not recommend using GSEA 3.0 or older versions</strong></li>
<li><strong>gsea_directory</strong> - full path to gsea results directory. Only populate if run_gsea is set to “no”.</li>
<li><strong>fdr_threshold</strong> - FDR threshold used to create enrichment map</li>
<li><strong>pval_thresho</strong> - pvalue threshold used to create enrichment map</li>
<li><strong>java_version</strong> - for backwards compatibility. For users using previous version of GSEA and older versions of java.</li>
</ol>
<p>For this initial analysis change:</p>
<ol style="list-style-type: decimal">
<li><strong>gsea_jar</strong> - change to the full path to the gsea jar that you were instructed to download in your pre-workshop setup instructions</li>
</ol>
<div class="rmd-caution">
<p>
This is not the same as the GSEA application that we used in Module 2 gsea lab. <img src="Module3/automation/images/gsea_command_jar.png" alt="open project" width="50%"/>
</p>
</div>
</div>
<div id="step-5---step-through-notebook-to-run-the-analysis" class="section level3">
<h3>Step 5 - Step through notebook to run the analysis</h3>
<p>The RNotebook is a mixture of markdown text and code blocks.</p>
<p>Read through the notebook to understand what each section is doing and sequentially run the code blocks by clicking on the play button at the top right of each code block.</p>
<p><img src="Module3/automation/images/rnotebook_play.png" alt="play" width="50%"/></p>
</div>
<div id="exercises" class="section level3">
<h3>Exercises</h3>
<p>Once you have run through the notebook and created your enrichment map automatically try the following:</p>
<ol style="list-style-type: decimal">
<li>change the fdr threshold and create a new network (<strong>without rerunning the whole notebook</strong>) with the lower FDR threshold.</li>
<li>change the similarity coeffecient and create a new network (<strong>without rerunning the whole notebook</strong>) with the lower FDR threshold.</li>
<li>re-run the notebook using the GSEA results you created on the first run without running GSEA.</li>
<li>Modify notebook to run with a different gmt file. (Downloaded from somewhere else or a different file found on <a href="http://download.baderlab.org/EM_Genesets/current_release/">baderlab genesets download site</a>)</li>
<li>Open the notebook Supplementary_Protocol5_Multi_dataset_theme_analysis.Rmd and run through it to create a multi dataset enrichment map.</li>
</ol>
</div>
<div id="additional-resources" class="section level3">
<h3>Additional resources</h3>
<p>Check out all the different notebooks available <a href="https://cytoscape.org/cytoscape-automation/for-scripters/R/notebooks/">here</a></p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="module-3-network-visualization-and-analysis-with-cytoscape.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="module-4-in-depth-analysis-of-networks-and-pathways.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": "https://github.com/risserlin/CBW_Pathways_bookdown/commit/master/3.1-Module3lab_gprofiler.Rmd",
"text": null
},
"view": {
"link": "https://github.com/risserlin/CBW_Pathways_bookdown/blob/master/3.1-Module3lab_gprofiler.Rmd",
"text": null
},
"download": ["CBW_Pathways.epub"],
"toc": {
"collapse": "section",
"scroll_highlight": true
}
});
});
</script>

</body>

</html>
